samples/zigbee/light_bulb/prj.conf

```diff
@@ -39,3 +39,10 @@ CONFIG_CRYPTO_NRF_ECB=y
 
 # Cryptocell is not supported through CSPRNG driver API: NCSDK-4813
 CONFIG_ENTROPY_CC310=n
+
+#Networking
+CONFIG_NET_IPV6_MLD=n
```

Comment:
When I disable `NET_IPV6` but `NET_NATIVE` is still enabled, I get error during net_pkt.c compilation:
`#error "Either IPv6 or IPv4 needs to be selected."`

When I disable both I get a lot of undefined references during linking to `net_pkt_*` stuff like `net_pkt_alloc_with_buffer`, `net_pkt_write`. These missing functions are needed in `ieee802154_nrf5.c`, `net_core.c`.

Reply:
That's unfortunate, we end up building parts of the net stack that have no use for Zigbee. Not sure if I have a good solution for this (at least w/o further modifications in Zephyr).

There's a special Kconfig called [NET_RAW_MODE](https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/CMakeLists.txt#L10) that disables literally everything in the net stack, beside `net_pkt`. Perhaps it could be used for Zigbee, but this config disables RX/TX queues as well (TX is not an issue since it's not used anyway), so it would be needed to re-implement `net_recv_data` function somewhere in the Zigbee area (same as some samples do in upstream Zephyr). I'll leave it up to you to decide if it's a viable solution for Zigbee.

Reply:
Re-implementing `net_recv_data` seems to be a cleaner solution, and PR adding probably Zigbee L2 could be closed.
However, considering solutions that involve parallel Thread and Zigbee, it would be better not to implement own `net_recv_data` but to use a common one instead, so that packets could be multiplexed in layers above `net_recv_data`.